scripts/mdpow-solvationenergy

```diff
@@ -179,37 +203,56 @@ def realpath_or_none(fn):
-                      metavar="NAME",
-                      help="solvent NAME for compound, 'water', 'octanol', or 'cyclohexane' [%default]")
-    parser.add_option('--ignore-corrupted', dest="permissive",
-                      action="store_true",
-                      help="skip lines in the md.xvg files that cannot be parsed. "
-                      "WARNING: Other lines in the file might have been corrupted in "
-                      "such a way that they appear correct but are in fact wrong. "
-                      "WRONG RESULTS CAN OCCUR! USE AT YOUR OWN RISK [%default]")
-    parser.set_defaults(plotfile="auto",
-                        energyfile="gsolv.txt",
-                        force=False, stride=10, permissive=False, solvent='water')
+    import argparse
+
+    parser = argparse.ArgumentParser(usage=__doc__)
+    parser.add_argument('-p', '--plotfile', dest="plotfile",
+                        help="plot dV/dlambda to FILE; use png or pdf suffix to "
+                        "determine the file type. 'auto' generates a pdf file "
+                        "DIRECTORY/dVdl_<molname>_<solvent>.pdf and 'None' disables it [%default]"
+                        "The plot function is only available for mdpow estimator,"
+                        "and is disabled when using alchemlyb estimators.",
+                        metavar="FILE")
+    parser.add_argument('--solvent', '-S', dest='solvent',
+                        choices=['water', 'octanol', 'cyclohexane'], metavar="NAME",
+                        help="solvent NAME for compound, 'water', 'octanol', or 'cyclohexane' [%default]")
+    parser.add_argument('-o', '--outfile', dest="outfile",
+                        help="append one-line results summary to FILE [%default]",
+                        metavar="FILE")
+    parser.add_argument('-e', '--energies', dest="energyfile",
+                        help="append solvation free energies to FILE [%default]",
+                        metavar="FILE")
+    parser.add_argument('--estimator', dest="estimator", default='alchemlyb',
+                        choices=['mdpow', 'alchemlyb'],
+                        help="choose the estimator to be used, alchemlyb or mdpow estimators")
+    parser.add_argument('--method', dest="method", default='MBAR',
+                        choices=['TI', 'MBAR', 'BAR'],
+                        help="choose the method to calculate free energy",)
+    parser.add_argument('--force', dest='force',
+                        action="store_true",
+                        help="force rereading all data [%default]")
+    parser.add_argument('--noSI', dest='SI',
+                        action="store_false",
+                        help="Disable statistical inefficiency analysis"
+                        "Statitical inefficiency analysis is performed by default when using"
+                        "alchemlyb estimators and is disabled when using mdpow estimator.")
```

Comment:
use the bool action

Reply:
I check the python documentation,  BooleanOptionalAction action option is added in 3.9, we can't use it in 2.7.